### PR TITLE
Sc-14487: Implement leveldb DeleteContact

### PR DESCRIPTION
**Pull request by DanielSollis**
### Scope of changes

Implements the DeleteContact method in store/leveldb

### Type of change

- [ ] bug fix
- [x] new feature
- [ ] documentation
- [ ] other (describe)

### Acceptance criteria

Describe how reviewers can test this change to be sure that it works correctly. Add a checklist if possible

### Author checklist

- [x] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [x]   Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [ ]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.

### CODE REVIEW

**Review by pdeziel — COMMENTED** [review]
Looks good, my comments here are pretty similar to the trtl PR so could you just make the same changes I requested there but in these methods?

File: pkg/models/v1/models.go [hunk 1]
```diff
@@ -223,10 +223,10 @@ func GetVASPEmailLog(vasp *pb.VASP) (emails []*EmailLogEntry, err error) {
 }
 
 // Normalize the email and convert to bytes
-func ContactKey(email string) []byte {
+func NormalizeEmail(email string) string {
```
Comment (pdeziel) [inline pkg/models/v1/models.go:226]: Is there a reason why this wouldn't be the same method as in the trtl store?

**Review by pdeziel — COMMENTED** [review]
Looks good, I just had a comment about the helper method.

File: pkg/models/v1/models.go (hunk 1 above)
Comment (DanielSollis) [reply pkg/models/v1/models.go:226]: The leveldb store requires a string to add a byte prefix to using the makeKey method with the preContacts string as an argument, while the trtl requires a byte key, I pulled out the common code into NormalizeEmail but it turns out that the last step is unfortunately different for each store.

Comment (pdeziel) [reply pkg/models/v1/models.go:226]: Gotcha, I definitely like having `NormalizeEmail` as a shared method. The only suggestion I would have is to move the call to `NormalizeEmail` to the `contactKey` method, which would simplify the code in the store methods.

Comment (DanielSollis) [reply pkg/models/v1/models.go:226]: That's a really good idea! I just implemented that and it's much cleaner.

**Pull request merged**